programs/drift/src/state/perp_market.rs

```diff
@@ -747,6 +749,44 @@ impl PerpMarket {
             default_min_auction_duration
         }
     }
+
+    pub fn get_mm_oracle_price_data(
+        &self,
+        oracle_price_data: OraclePriceData,
+        clock_slot: u64,
+        oracle_guard_rails: &ValidityGuardRails,
+    ) -> DriftResult<MMOraclePriceData> {
+        let delay = clock_slot
+            .cast::<i64>()?
+            .safe_sub(self.amm.mm_oracle_slot.cast::<i64>()?)?;
+        let oracle_data = OraclePriceData {
+            price: self.amm.mm_oracle_price,
+            delay,
+            confidence: oracle_price_data.confidence,
+            has_sufficient_number_of_data_points: true,
+        };
+        let oracle_validity = if self.amm.mm_oracle_price == 0 {
+            OracleValidity::NonPositive
+        } else {
+            oracle_validity(
+                MarketType::Perp,
+                self.market_index,
+                self.amm.historical_oracle_data.last_oracle_price_twap,
+                &oracle_data,
+                &oracle_guard_rails,
+                self.get_max_confidence_interval_multiplier()?,
+                &self.amm.oracle_source,
+                true,
```

Comment:
should change this to an enum `log_mode` and then change logs for MM oracle so it's easier to debug